lib/config/decrypt.ts

```diff
@@ -209,3 +228,29 @@ Refer to migration documents here: https://docs.renovatebot.com/mend-hosted/migr
   logger.trace({ config: decryptedConfig }, 'decryptedConfig');
   return decryptedConfig;
 }
+
+function getAzureCollection(): string | undefined {
+  const platform = GlobalConfig.get('platform');
+  if (platform !== 'azure') {
+    return undefined;
+  }
+
+  const endpoint = GlobalConfig.get('endpoint');
+  const endpointUrl = parseUrl(endpoint);
+  if (endpointUrl === null) {
+    // should not happen
+    logger.warn({ endpoint }, 'Unable to parse endpoint for token decryption');
+    return undefined;
+  }
+
+  const azureCollection = endpointUrl.pathname
+    ?.split('/')
+    .filter(is.nonEmptyString)
+    .join('/');
```

Comment:
why do we need to split and join here at all?

Reply:
to remove the slash at the start and at the end
e.g
* `https://dev.azure.com/` the URL pathname holds `/`
* `https://dev.azure.com/aaa/bbb/` the URL pathname holds `/aaa/bbb/`

Reply:
use `trimSlashes` url util? i think `endpointUrl.pathname` should never undefined or null here?